pkg/sql/lease.go

```diff
@@ -587,6 +600,43 @@ func (t *tableState) acquire(
 	return t.findForTimestamp(ctx, timestamp, m)
 }
 
+// acquireAlways ensures a new lease is acquired. This is used for making sure a
+// new lease moves forward when refreshing the lease.
+func (t *tableState) acquireAlways(ctx context.Context, m *LeaseManager) error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	// Block and wait for the new lease to be acquired on another routine to
+	// prevent a race condition of entering acquireNodeLease
+	t.acquireWait()
+
+	// Ensure a lease is acquired.
+	return t.acquireNodeLease(ctx, m, hlc.Timestamp{})
+}
+
+// startRefreshingRoutine begins the goroutine responsible for refreshing a
+// table lease.
+// TODO(joey): Currently this routine will run forever for each table
+// descriptor. Ideally, we stop the routine if the table descriptor hasn't been
+// used recently.
+//
+// t.mu needs to be locked.
+func (t *tableState) startRefreshingRoutine(m *LeaseManager) {
+	ctx := context.TODO()
```

Comment:
just pass in the context

Reply:
I mentioned this in the last comment, but it turned out that that was failing (it wasn't evident until I was testing cockroach -- I'm not sure how this could be testable). It's because when the refreshing routine goes to use the context (several minutes later) the context has been canceled.

Reply:
Okay leave it as is